### PR TITLE
Matts post picker modifications

### PR DESCRIPTION
**Pull request by mattheu**
I've been reviewing this PR and getting it to work on my project. This is what I've changed: 

* Renamed `.jsx` to `.js` so that it actually builds. 
* Fix typo in index.js
* Support for multiple taxonomies. I ran into some issues with taxonomies like post tag where the taxonomy slug `post_tag` is different from the `rest_base` (`tags`) and the singular name (`Tag`). 
* Move the actual `getEntityRecords` call into `PostList`, so that the query is done only when opening the modal, and not on initial page load. 
* Reduce the number of queried posts to `30` instead of `-1`. Unbounded queries are actually capped at 100 anyway.  But seems I've got some unrelated performance issues on my local dev site and this was taking way too long. Long term... would love pagination but as an MVP I don't think it's necessary given you can search/filter.

### CODE REVIEW

File: src/components/PostPicker/index.js [hunk 1]
```diff
@@ -49,8 +63,8 @@ function PostList( props ) {
 						queriedPosts.map( post => (
 							<CheckboxControl
 								key={ post.id }
-								checked={ values?.includes( post.id ) }
-								label={ post.title?.rendered }
+								checked={ values.includes( post.id ) }
+								label={ post.title?.rendered || __( '(No title)', 'block-editor-components' ) }
```
Comment (tomjn) [inline src/components/PostPicker/index.js:67]: Things are nicer with spaces

```suggestion
								label={ post.title?.rendered || __( '( No title )', 'block-editor-components' ) }
```

File: src/components/TermSelector/index.js [hunk 2]
```diff
@@ -47,26 +47,27 @@ function generateIdToTitleMap( termObjects ) {
...
 			taxonomyTermsByTitle,
 		};
-
 	}, [ taxonomy ] );
 
 	const selectedTerms = value.map( id => taxonomyTermsById[id] );
 
 	return ( <FormTokenField
-		label={ `Filter by ${taxonomy}` }
+		label={ `Filter by ${ taxObject ? taxObject.labels.singular_name : '' }` }
```
Comment (tomjn) [inline src/components/TermSelector/index.js:70]: should this be translatable?

**Review by tomjn — CHANGES_REQUESTED** [review]
2 changes but otherwise a good improvement to the PR

File: src/components/TermSelector/index.js (hunk 2 above)
Comment (mattheu) [reply src/components/TermSelector/index.js]: Good catch. Fixed.

**Review by tomjn — APPROVED** [review]
👍🏻 good changes

**Comment by mattheu** [comment]
As @tomjn approved this, going to merge into Joe's feature branch.

**Pull request merged**